include/seqan3/range/view/interleave.hpp

```diff
@@ -0,0 +1,383 @@
+        }
+    }
+};
+
+} // namespace seqan3::detail
+
+// ============================================================================
+//  view::interleave (adaptor instance definition)
+// ============================================================================
+
+namespace seqan3::view
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief A view that interleaves a given range into another range at regular intervals.
+ * \tparam urng_t The type of the range being processed.
+ * \tparam inserted_rng_t The type of the range being inserted.
+ * \param[in] urange The range being processed.
+ * \param[in] inserted_range The range being inserted.
+ * \param[in] step_size A value of size_type which indicates the interval to insert the inserted_range.
+ * \returns A range with the second range inserted at regular intervals. See below for properties of said range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to insert one range into another range at regular intervals. It behaves essentially like
+ * `| std::view::chunk(step_size) | std::view::join(inserted_range)` except that for input that models
+ * std::ranges::RandomAccessRange and std::ranges::SizedRange a more efficient data structure is returned
+ * (otherwise it returns exactly the above combination of views).
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)  |
+ * |---------------------------------|:-------------------------------------:|:-------------------------------:|
+ * | std::ranges::InputRange         | *required*                            | *preserved*                     |
+ * | std::ranges::ForwardRange       | *required*                            | *preserved*                     |
+ * | std::ranges::BidirectionalRange | *required*                            | *preserved*                     |
+ * | std::ranges::RandomAccessRange  | *required*                            | *preserved*                     |
+ * | std::ranges::ContiguousRange    |                                       | *lost*                          |
+ * |                                 |                                       |                                 |
+ * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                    |
+ * | std::ranges::View               |                                       | *guaranteed*                    |
+ * | std::ranges::SizedRange         | *required*                            | *preserved*                     |
+ * | std::ranges::CommonRange        |                                       | *preserved*                     |
+ * | std::ranges::OutputRange        |                                       | *preserved*                     |
+ * | seqan3::const_iterable_concept  |                                       | *preserved*                     |
```

Comment:
```suggestion
 * | seqan3::ConstIterableRange      |                                       | *preserved*                     |
```

This was recently renamed by @MitraDarja.... Also beloew